### PR TITLE
Menu

### PR DESCRIPTION
**Pull request by Apatil15**
Added new menu component with the supportive following components.

1. Menu
2. MenuItem
3. MenuDivider
4. MenuHeader
5. MenuGroup

Created folders according to menu component.
1. _react/src/components/BaseMenu/_
2. _react/src/components/Menu/_

Also created following style file and token
_core/src/components/menu/index.scss_ 
_/core/src/components/menu/tokens.scss_

Taking care design menu as per the discussion. 
 
Following features has been added

1. Menu item hover color.
2. Menu item focus color.
3. Menu item selected color.
4. On menu item click get all the selected properties.
5. Auto hide when click on external element.
6. Set element position as per the parent element.
7. Set default focus.
8. Dynamic json menu generation.
9. Menu item click event
10.Menu item url redirect

Unit testing is remaining I am working on it.

@sh0ji Please review code and let me know your review comments.

### CODE REVIEW

**Review by sh0ji — CHANGES_REQUESTED** [review]
Thanks for this, @Apatil15! This is probably our most complex component yet so I appreciate everything that went into it.

I haven't reviewed everything in its entirety but I see some things that are probably best addressed now rather than being comprehensive.

1. `Base` components should really only handle polyfilling behavior and setting the unstyled semantics of a pattern. In other words, they should mostly just inherit an HTML interface as their props and default props should only ever be used for polyfilling. This effectively means that some of your code in the `Base` components needs to be pulled up to the non-`Base` ones.
   - I outlined what I think `BaseMenuGroup` should do in a code comment so you can see what I mean.
2. Menus cannot be used to select items. That's what a [listbox](https://w3c.github.io/aria/#listbox) is used for (and it's what our `BaseListbox` + `BaseOption` accomplishes).
   - The exception to this might be a `menuitemcheckbox` or `menuitemradio` but those are out of scope for this initial PR.
3. It occurs to me that we'll need a `Link` or `Anchor` component to address the issue of router-provided links. I'll open a separate issue for that (it's long been neglected).

File: packages/react/src/components/BaseMenu/baseMenuDivider.tsx [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import React from 'react';
+
+export const BaseMenuDivider = React.forwardRef<HTMLElement>(({
+	...props
```
[the remark this answers is not in the document]
Comment (Apatil15) [reply packages/react/src/components/BaseMenu/baseMenuDivider.tsx]: Resolved as per the comments

File: packages/react/src/components/BaseMenu/baseMenuDivider.tsx [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+}) => (
+	<li
+		className='nds-menu-divider'
```
[the remark this answers is not in the document]
Comment (Apatil15) [reply packages/react/src/components/BaseMenu/baseMenuDivider.tsx]: Style applied as per the role

File: packages/react/src/components/BaseMenu/baseMenuGroup.tsx [hunk 3]
```diff
@@ -0,0 +1,16 @@
+import React from 'react';
+import { BaseMenuDivider } from './baseMenuDivider';
+
+export interface BaseMenuGroupProps extends React.LiHTMLAttributes<HTMLLIElement> {
+	ariaLabel?:string,
+}
+
+export const BaseMenuGroup = React.forwardRef<HTMLLIElement, BaseMenuGroupProps>(({
```
[the remark this answers is not in the document]
Comment (Apatil15) [reply packages/react/src/components/BaseMenu/baseMenuGroup.tsx]: Code has been update

File: packages/react/src/components/BaseMenu/baseMenuDivider.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (Apatil15) [reply packages/react/src/components/BaseMenu/baseMenuDivider.tsx]: Created new menuButton component and inherited from button component

**Comment by sh0ji** [comment]
We'll address this in another PR.

**Pull request closed without merging**